js_modules/dagit/packages/core/src/runs/LogsProvider.tsx

```diff
@@ -159,12 +162,29 @@ const useLogsProviderWithSubscription = (runId: string) => {
   // Batch the nodes together so they don't overwhelm the animation of the Gantt,
   // which depends on a bit of a timing delay to maintain smoothness.
   const throttledSetNodes = React.useMemo(() => {
-    return throttle((hasMore: boolean, cursor: string) => {
-      const queued = [...queue.current];
+    return throttle(() => {
+      if (!queue.current.length) {
+        return;
+      }
+      const queuedLogs = [...queue.current];
       queue.current = [];
-      dispatch({type: 'append', queued, hasMore, cursor});
+      const queuedMessages = ([] as RunDagsterRunEventFragment[]).concat(
+        ...queuedLogs.map((log) => log.messages),
+      );
+      const lastLog = queuedLogs[queuedLogs.length - 1];
+      const hasMore = lastLog.hasMorePastEvents;
+      const cursor = lastLog.cursor;
+
+      dispatch({type: 'append', queued: queuedMessages, hasMore, cursor});
+      const nextPipelineStatus = pipelineStatusFromMessages(lastLog.messages);
+
+      // If we're still loading past events, don't sync to the cache -- event chunks could
+      // give us `status` values that don't match the actual state of the run.
+      if (nextPipelineStatus && !hasMore) {
+        syncPipelineStatusToApolloCache(nextPipelineStatus);
+      }
```

Comment:
This may have been a bug before, but I feel like there's a weird edge case here in which the "last pipeline status event" is NOT in the last "batch" seen by `throttle`, which means these two arguments are never `true` at the same time and the latest pipeline status is not set?

It might be better to say `if (!hasMore) { syncPipelineStatusToApolloCache(<<all messages>>) }` to set the pipeline status instead of just considering the last batch.